src/widgets/header/ui/Profile.tsx

```diff
@@ -0,0 +1,31 @@
+import { Avatar, IconButton, Tooltip } from '@mui/material';
+import { useState } from 'react';
+import { UserMenu } from './UserMenu';
+
+interface ProfileProps {
+  user: {
+    name?: string | null;
+    image?: string | null;
+  };
+}
+
+export const Profile = ({ user }: ProfileProps) => {
+  const [anchorElUser, setAnchorElUser] = useState<HTMLElement | null>(null);
+
+  const handleOpenUserMenu = (event: React.MouseEvent<HTMLElement>) => {
```

Comment:
![image](https://user-images.githubusercontent.com/20379120/216723749-97a94553-aa7a-4399-8481-a69eeee6f323.png)

Reply:
Do you think wrapping in useCallback and memoizing component that accepts this function as prop will be more efficient?